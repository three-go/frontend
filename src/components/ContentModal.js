import React from "react";

import { Modal, StyleSheet, Text, View, ScrollView } from "react-native";

const ContentModal = ({ title, content, children, isVisible }) => {
  return (
    <View style={styles.container}>
      <Modal animationType="fade" transparent={true} visible={isVisible}>
        <View style={styles.container}>
          <View style={styles.modalContainer}>
            <Text style={styles.title}>{title}</Text>
            <ScrollView style={styles.contentContainer}>
              <Text>{content}</Text>
            </ScrollView>
            <View style={styles.buttonContainer}>{children}</View>
          </View>
        </View>
      </Modal>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    marginTop: 22,
  },
  modalContainer: {
    alignItems: "center",
    width: "80%",
    height: "70%",
    margin: 20,
    padding: 35,
    borderRadius: 20,
    backgroundColor: "white",
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 5,
  },
  title: {
    marginBottom: 15,
    textAlign: "center",
    fontSize: 25,
    fontWeight: "bold",
  },
  contentContainer: {
    flex: 1,
  },
  buttonContainer: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-evenly",
    width: "100%",
    marginTop: 22,
  },
});

export default ContentModal;
