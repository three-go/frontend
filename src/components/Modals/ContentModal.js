import React from "react";

import { Modal, StyleSheet, Text, View } from "react-native";
import PropTypes from "prop-types";

import Carousel from "../Layouts/Carousel";
import { colors } from "../../common/constants";

const ContentModal = ({
  isVisible,
  title,
  content,
  handleRenderScoreItem,
  size,
  children,
}) => {
  return (
    <Modal animationType="fade" transparent={true} visible={isVisible}>
      <View style={styles.container}>
        <View style={styles.modalContainer}>
          <Text style={styles.title}>{title}</Text>
          <View style={styles.contentContainer}>
            <Carousel
              content={content}
              renderItem={handleRenderScoreItem}
              size={size}
            />
          </View>
          <View style={styles.buttonContainer}>{children}</View>
        </View>
      </View>
    </Modal>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    marginTop: 22,
  },
  modalContainer: {
    alignItems: "center",
    width: "80%",
    height: "70%",
    margin: 20,
    padding: 35,
    borderWidth: 5,
    borderRadius: 20,
    borderColor: colors.ivory,
    backgroundColor: colors.gray,
    shadowColor: colors.black,
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 5,
  },
  title: {
    marginBottom: 15,
    color: colors.ivory,
    textAlign: "center",
    fontSize: 25,
    fontWeight: "bold",
  },
  contentContainer: {
    flex: 1,
    width: 300,
    backgroundColor: colors.gray,
  },
  buttonContainer: {
    flexDirection: "row",
    justifyContent: "space-evenly",
    alignItems: "center",
    width: "100%",
    marginTop: 22,
  },
});

export default ContentModal;

ContentModal.propTypes = {
  isVisible: PropTypes.bool.isRequired,
  title: PropTypes.string.isRequired,
  content: PropTypes.arrayOf(
    PropTypes.oneOfType([
      PropTypes.shape({
        children: PropTypes.array,
        title: PropTypes.string,
        titleContent: PropTypes.string,
      }),
      PropTypes.arrayOf(
        PropTypes.shape({
          id: PropTypes.string,
          name: PropTypes.string,
          score: PropTypes.number,
        })
      ),
    ])
  ).isRequired,
  handleRenderScoreItem: PropTypes.func,
  size: PropTypes.shape({
    GAP: PropTypes.number,
    OFFSET: PropTypes.number,
    PAGE_WIDTH: PropTypes.number,
    SCREEN_WIDTH: PropTypes.number,
  }).isRequired,
  children: PropTypes.node.isRequired,
};
