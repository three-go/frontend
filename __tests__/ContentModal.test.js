import React from "react";
import { render } from "@testing-library/react-native";
import "@testing-library/react-native";
import { Text, View } from "react-native";
import ContentModal from "../src/components/Modals/ContentModal";

jest.mock("react-native-sound", () => {
  const Sound = () => {
    const obj = {};
    obj.sound = "music";
    return obj;
  };
  return Sound;
});

const mockData = {
  title: "test title",
  content: [
    { title: "Title1", description: "description1" },
    { title: "Title2", description: "description2" },
  ],
  handleRenderScoreItem: ({ item }) => {
    return (
      <View>
        <Text>{item.title}</Text>
        <Text>{item.description}</Text>
      </View>
    );
  },
  size: {
    SCREEN_WIDTH: 300,
    GAP: 15,
    OFFSET: 15,
    PAGE_WIDTH: 300 - (15 + 15) * 2,
  },
};

it("ContentModal should render correctly", () => {
  const component = render(
    <ContentModal isVisible={true} {...mockData}>
      <Text>test children</Text>
    </ContentModal>
  );

  expect(component).toBeTruthy();

  const { getByText } = render(
    <ContentModal isVisible={false} {...mockData}>
      <Text>test children</Text>
    </ContentModal>
  );

  expect(() => getByText(mockData.title)).toThrow(
    "Unable to find an element with text: test title"
  );
});

it("ContentModal should render title correctly", () => {
  const { getByText } = render(
    <ContentModal isVisible={true} {...mockData}>
      <Text>test children</Text>
    </ContentModal>
  );

  expect(getByText(/test title/i)).toHaveTextContent("test title");
});

it("ContentModal should have content property", () => {
  const { getByText, toJSON } = render(
    <ContentModal isVisible={true} {...mockData}>
      <Text>test children</Text>
    </ContentModal>
  );
  const carouselJson = toJSON();

  expect(getByText(/Title1/i)).toHaveTextContent("Title1");
  expect(getByText(/Title1/i)).toBeTruthy();
  expect(getByText(/description1/i)).toBeTruthy();
  expect(getByText(/Title2/i)).toBeTruthy();
  expect(getByText(/description2/i)).toBeTruthy();
  expect(carouselJson).toMatchSnapshot();
  expect(carouselJson).toBeTruthy();
});

it("ContentModal should render children correctly", () => {
  const { getByText } = render(
    <ContentModal isVisible={true} {...mockData}>
      <Text>test children</Text>
    </ContentModal>
  );

  expect(getByText(/test children/i)).toHaveTextContent("test children");
});
