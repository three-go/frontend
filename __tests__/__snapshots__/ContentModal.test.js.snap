// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentModal should have content property 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "marginTop": 22,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#96A1A8",
          "borderColor": "#FCF8F6",
          "borderRadius": 20,
          "borderWidth": 5,
          "elevation": 5,
          "height": "70%",
          "margin": 20,
          "padding": 35,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 4,
          "width": "90%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FCF8F6",
            "fontSize": 25,
            "fontWeight": "bold",
            "marginBottom": 15,
            "textAlign": "center",
          }
        }
      >
        test title
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "#96A1A8",
            "flex": 1,
            "width": 300,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "height": "60%",
              "justifyContent": "center",
            }
          }
        >
          <RCTScrollView
            automaticallyAdjustContentInsets={false}
            contentContainerStyle={
              Object {
                "paddingHorizontal": 22.5,
              }
            }
            data={
              Array [
                Object {
                  "description": "description1",
                  "title": "Title1",
                },
                Object {
                  "description": "description2",
                  "title": "Title2",
                },
              ]
            }
            decelerationRate="fast"
            getItem={[Function]}
            getItemCount={[Function]}
            horizontal={true}
            keyExtractor={[Function]}
            onContentSizeChange={[Function]}
            onLayout={[Function]}
            onMomentumScrollBegin={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            pagingEnabled={true}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            showsHorizontalScrollIndicator={false}
            snapToAlignment="start"
            snapToInterval={255}
            stickyHeaderIndices={Array []}
            viewabilityConfigCallbackPairs={Array []}
          >
            <View>
              <View
                onLayout={[Function]}
                style={
                  Array [
                    Object {
                      "flexDirection": "row",
                    },
                    null,
                  ]
                }
              >
                <View>
                  <Text>
                    Title1
                  </Text>
                  <Text>
                    description1
                  </Text>
                </View>
              </View>
              <View
                onLayout={[Function]}
                style={
                  Array [
                    Object {
                      "flexDirection": "row",
                    },
                    null,
                  ]
                }
              >
                <View>
                  <Text>
                    Title2
                  </Text>
                  <Text>
                    description2
                  </Text>
                </View>
              </View>
            </View>
          </RCTScrollView>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-evenly",
            "marginTop": 22,
            "width": "100%",
          }
        }
      >
        <Text>
          test children
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;
